src/Symfony/Bridge/Doctrine/Form/EventListener/MergeDoctrineCollectionListener.php

```diff
@@ -27,15 +27,31 @@
  */
 class MergeDoctrineCollectionListener implements EventSubscriberInterface
 {
+    // Keeps BC. To be removed in 4.0
+    private $bc = true;
+
     public static function getSubscribedEvents()
     {
         // Higher priority than core MergeCollectionListener so that this one
         // is called before
-        return array(FormEvents::SUBMIT => array('onBind', 10));
+        return array(
+            FormEvents::SUBMIT => array(
+                // BC
+                array('onBind', 10),
+                array('onSubmit', 5),
+            ),
+        );
     }
 
-    public function onBind(FormEvent $event)
+    public function onSubmit(FormEvent $event)
     {
+        // If onBind() is overridden then logic has been executed
+        if ($this->bc) {
```

Comment:
Not sure if that is sufficient. Someone who overrode the `onBind()` method will likely also have added a call to `parent::onBind()`.